tests/test_alias.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2024, Basalte bv
+
+import os
+
+import pytest
+
+from conftest import cmd
+
+assert 'TOXTEMPDIR' in os.environ, "you must run these tests using tox"
+
+@pytest.fixture(autouse=True)
+def autouse_tmpdir(config_tmpdir, west_init_tmpdir):
+    # Since this module tests west's configuration file features,
+    # adding autouse=True to the config_tmpdir and west_init_tmpdir fixtures
+    # saves typing and is less error-prone than using it below in every test case.
+    pass
+
+def test_alias_commands():
+    cmd('config alias.test1 topdir')
+    cmd('config --global alias.test2 topdir')
+    cmd('config --system alias.test3 topdir')
+
+    topdir_out = cmd('topdir')
+
+    assert cmd('test1') == topdir_out
+    assert cmd('test2') == topdir_out
+    assert cmd('test3') == topdir_out
+
+def test_alias_help():
+    cmd('config alias.test topdir')
+
+    assert cmd('-h test') == cmd('help test')
+
+def test_alias_recursive_commands():
+    cmd('config alias.test1 topdir')
+    cmd('config alias.test2 test1')
```

Comment:
Can you please add some whitespace and quoting in the recursive test case to make it more demanding? Maybe 3 levels too. Basically: something that breaks when you remove the `shlex` parsing that you just added.

It shouldn't require much more test code.

Reply:
Also: some infinite loop as a "negative" test?